### PR TITLE
Add tensorflow data validation tests

### LINKED ISSUE #8: Set up data validation

### PR DESCRIPTION
**Pull request by adondera**
Adds TFDV as a dev-dependency and runs barebones schema validation tests for the raw files.

Closes #8

### CODE REVIEW

File: tests/test_validate_schema.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    dataset = pd.read_csv(f"{DATA_PATH}/{filename}.tsv", sep='\t')
+    schema = tfdv.load_schema_text(SCHEMA_PATH)
+    stats = tfdv.generate_statistics_from_dataframe(dataset)
+    anomalies = tfdv.validate_statistics(stats, schema)
+    return anomalies
+
+
+def test_train_schema():
+    train_anomalies = get_anomalies('train')
+    assert len(train_anomalies.anomaly_info) == 0
```
Comment (wandersiemers) [inline tests/test_validate_schema.py]: You could do truth value testing, which pep8 recommends if `anomaly_info` evaluates to false when it is empty, but I'm not sure it is.

File: tests/test_validate_schema.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def test_validation_schema():
+    validation_anomalies = get_anomalies('validation')
+    assert len(validation_anomalies.anomaly_info) == 0
```
Comment (wandersiemers) [inline tests/test_validate_schema.py]: Same here

File: assets/data/schema.pbtxt [hunk 3]
```diff
@@ -0,0 +1,26 @@
+feature {
+  name: "title"
+  type: BYTES
```
Comment (V0XNIHILI) [inline assets/data/schema.pbtxt]: Shouldnt the type be `STRING`?
image

File: assets/data/schema.pbtxt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  shape {
+    dim {
+      size: 1
+    }
+  }
+}
+feature {
+  name: "tags"
+  type: BYTES
```
Comment (V0XNIHILI) [inline assets/data/schema.pbtxt]: Same here

File: tests/test_validate_schema.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def test_test_schema():
+    test_anomalies = get_anomalies('test')
+    assert len(test_anomalies.anomaly_info) == 1
```
Comment (V0XNIHILI) [inline tests/test_validate_schema.py]: Why == 1?

**Review by V0XNIHILI — CHANGES_REQUESTED** [review]

File: tests/test_validate_schema.py (hunk 5 above)
Comment (adondera) [reply tests/test_validate_schema.py]: The only anomaly is the missing `label` column and there aren't really any other types of anomalies that can occur. I honestly don't really see the point of the test set. I added this test so that there is a test that case actually detects an anomaly.

I can change it to something like `>=1`, or something similar, but I can't come up with an example where this would actually be better. Do you have a better assertion in mind?

File: assets/data/schema.pbtxt (hunk 3 above)
Comment (adondera) [reply assets/data/schema.pbtxt]: The `infer_schema` function has `max_string_domain_size: int = 100`. I don't know enough about the fields of the schema proto to know if it's a good idea to increase this enough for the fields to be classified as strings.

File: assets/data/schema.pbtxt (hunk 4 above)
Comment (adondera) [reply assets/data/schema.pbtxt]: Same thing as before, this is the schema inferred by TFDV.

File: tests/test_validate_schema.py (hunk 1 above)
Comment (adondera) [reply tests/test_validate_schema.py]: Done, nice suggestion 👍🏻

File: tests/test_validate_schema.py (hunk 2 above)
Comment (adondera) [reply tests/test_validate_schema.py]: Done

File: tests/test_validate_schema.py (hunk 5 above)
Comment (V0XNIHILI) [reply tests/test_validate_schema.py]: I agree,  there's not much point here. but maybe for cleanness you can. make a separate schema for the test file, just deleting the tags feature basically?

**Comment by V0XNIHILI** [comment]
Probably need to merge main into this as well?

File: tests/test_validate_schema.py (hunk 5 above)
Comment (adondera) [reply tests/test_validate_schema.py]: Done

**Review by V0XNIHILI — APPROVED** [review]

**Comment by farsene** [comment]
Can you remove the empty test.py file?

**Review by farsene — APPROVED** [review]
